### PR TITLE
Checks for possible duration + start time

### LINKED ISSUE #175: Duration over the current day
As per the description, if we have a duration such as 999 that exceeds the day itself, it will cause alot of bugs everywhere. So I suggest having a check such that the start time + duration is < the end of the day ah.
{
      "type" : "JsonAdaptedSession",
      "sessionDate" : "2021-01-01T20:00",
      "duration" : "999",
      "subject" : "Biology",
      "fee" : "80.0"
    }

### PR DESCRIPTION
**Pull request by enhao25**
Resolves #175

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddSessionCommand.java [hunk 1]
```diff
@@ -67,10 +66,10 @@ public CommandResult execute(Model model) throws CommandException {
...
                 throw new CommandException(SESSION_OVERLAP);
             }
-        } else {
-            if (model.hasOverlappingSession(sessionToAdd)) {
-                throw new CommandException(SESSION_OVERLAP);
-            }
+        } else if (model.hasOverlappingSession(sessionToAdd)) {
+            throw new CommandException(SESSION_OVERLAP);
+        } else if (!sessionToAdd.isPossibleEndTime()) {
+            throw new CommandException(Session.MESSAGE_CONSTRAINTS);
```
Comment (chwoozy) [inline src/main/java/seedu/address/logic/commands/AddSessionCommand.java]: Just checking if this should be caught earlier on, before checking if overlap.

**Review by chwoozy — COMMENTED** [review]
Got a comment on the checking of session range.

File: src/main/java/seedu/address/logic/commands/AddSessionCommand.java (hunk 1 above)
Comment (enhao25) [reply src/main/java/seedu/address/logic/commands/AddSessionCommand.java]: O.o actually I made some updates, and just check during the parser method before the session is even created ah. See if this version is ok.

**Review by nowknowing — APPROVED** [review]
All LGTM

**Review by chwoozy — APPROVED** [review]
LGTM! ☕️

**Pull request merged**